### DISCUSSION TITLE
Prefix on transcript for stats of subtitle used

### DISCUSSION THREAD

**Discussion by Marzal** [skymethod/op3#16] [Ideas]
Right now we have 2 feeds for the same podcast, one with de audio in [ogg](https://kdeexpress.gitlab.io/feed) and another in [mp3](https://kdeexpress.gitlab.io/feed_mp3), but the same guid. Not sure if OP3 detects this and mix the downloads stats, but thinking about that we came with a question.

What would happen if we also put the prefix in the `<podcast:transcript url="` would this mess things up. It would be recorded but not used?

It's a crazy idea or just out of scope to use OP3 to also track transcripts "usage"?. We think it could be a good measure of the adoption of the tag.

PD: Thanks for this great service.

**Comment by johnspurlock-skymethod** [comment]
Hi David, which one is the correct version of your show? 

I see:
https://kdeexpress.gitlab.io/posts/kdeexpress/index.xml
and
https://kdeexpress.gitlab.io/feed_mp3

As long as the urls are associated to one and only one item in the feed, they should roll up to the show successfully.

The OP3 prefix will redirect any link, but it is primarily intended for podcast episode redirects (hence the "e" in the prefix).  However, some have used it for live streams, which is fine since it's also an artifact in the public rss feed.

Transcripts to me fall within the category of things that I'd be fine with you experimenting with if it helps you.  Just make sure the prefix url resides in a public rss feed.

You won't be able to tie transcripts to "downloads", but you can certainly use the "redirect logs" api to keep track of how often they are requested, from where, and from which user agent.

In general, I stop redirects working from most other uses of the open redirect service, because spammers love them - but I will make sure transcripts go through fine.

**Comment by Marzal** [comment]
https://kdeexpress.gitlab.io/posts/kdeexpress/index.xml
Is the feed that Hugo generates. It has 2 enclosures for each item (ogg + mp3)

Then we split this xml in two
https://kdeexpress.gitlab.io/feed with only the enclosures with the ogg and
https://kdeexpress.gitlab.io/feed_mp3 with only enclosures with mp3
They are identical (same guids..) except por the enclosures with the differents extension.

We are in our way to implement the alternate enclosures and fix the atom rel link to point right url, given that now all 3 xml point to the first one with 2 enclosures.

I will have a look at the API to see if I'm able to make use of it (I'm a sysadmin with limited dev skills 😁)

Thanks for the input and for OP3 in general.

**Comment by johnspurlock-skymethod** [comment]
I see, bear in mind most players do not support multiple `<enclosure>` elements under a single item, and will ignore one of them.   Let me know when you get the feed into a shape you want to serve as your canonical feed for this show, and I'll make sure OP3 points to it.

The OP3 api to query redirects should not be too daunting.  Start here https://op3.dev/api/docs#tag/redirect-logs/operation/queryRedirectLogs and you can add the `url` query parameter to limit the results to all redirect requests for that url, you can plug your transcript url in there once you add the prefix.

**Reply by Marzal** [reply]
Yes, most apps take the first item, some take the second one if the first is not in a supported format, and a few apps fail if the first enclosure is not in a supported format, the users of my feed generator are aware and want that behavior and have the feed_mp3 for the apps that only support one enclosure or only mp3.

I prefer to respect the spec and have 1 feed with only ogg and 1 with only mp3.

By the way I have started playing with the API and have a few question, I'm opening another discussion in case anyone has the same doubts. Sorry for the trouble.
